ported/libibverbs/rc_pingpong.c

```diff
@@ -508,11 +508,15 @@ int main(int argc, char *argv[])
 	hints->ep_attr->type = FI_EP_MSG;
 	hints->caps = FI_MSG;
 	hints->mode = FI_LOCAL_MR;
-	hints->addr_format = FI_SOCKADDR;
 
-	rc = ft_read_addr_opts(&node, &service, hints, &flags, &opts);
-	if (rc)
-		return 1;
+	if (!opts.dst_node) {
+		flags |= FI_SOURCE;
+		node = opts.src_node;
+		service = opts.src_port;
+	} else {
+		node = opts.dst_node;
+		service = opts.dst_port;
+	}
```

Comment:
Not a big issue but can we put the common code for setting node, service, and flags in ft_read_addr_opts()? Every example seems to have the same code.